### ISSUE TITLE
Will the 'backward' method in 'LSTM.lua' override the default 'backward' method in 'nn.Module'?

### ISSUE THREAD

**Issue by crazylyf** [jcjohnson/torch-rnn#87]
Dear Author,

I have noticed that there is a 'backward' method in the package 'LSTM.lua'. I'm wondering whether it will override the same methods in its parent package 'nn.Module'. If true, the method 'updateGradInput' and 'accGradParameters' will never be called, and the gradient will not update correctly. Is my understanding right?

Best

**Comment by jcjohnson** [comment]
The `backward` method in LSTM.lua does override the `backward` method in `nn.Module`. The gradients will be fine, because the LSTM `backward` method both computes gradients with respect to inputs and accumulates gradients with respect to parameters.

This is similar to the `backward` method of `nn.Sequential`: https://github.com/torch/nn/blob/master/Sequential.lua#L78

In general it is better to override `updateGradInput` and `accGradParameters`, but for RNNs these two methods share a lot of computation so it is more efficient to combine them into one method.
